src/components/dropdown/dropdown.controller.js

```diff
@@ -132,19 +137,48 @@ export default class ScDropdownCtrl {
     }
 
     _positionVerticalMenu() {
-        const positionAction = this.trigger[0].getBoundingClientRect();
-        const positionMenu = this.content[0].getBoundingClientRect();
-        let menuTopPosition = positionAction.bottom + CARRET_HEIGHT;
+        const blocAction = this.trigger[0];
+        const blocContent = this.content[0];
+        const positionAction = blocAction.getBoundingClientRect();
+        const positionContent = blocContent.getBoundingClientRect();
+        const windowSize = this.window[0].innerHeight;
 
-        //when menu bottom is outside of the window, we position the menu at the top of the button
-        if ((positionAction.bottom +  positionMenu.height + CARRET_HEIGHT)> this.window[0].innerHeight) {
-            menuTopPosition = positionAction.top - CARRET_HEIGHT - positionMenu.height;
-            this.content.addClass('top');
+
+
+
+        //display the menu where there is most space top or bottom
+        const freeSpaceBottom = windowSize - positionAction.bottom;
+        const freeSpaceTop = positionContent.top;
+        const heightContent = positionContent.height;
+        let offsetBorder = parseInt(this.distanceFromBorder);
+
+        if (isNaN(offsetBorder)){
+            offsetBorder = 30;
         }
-        else {
+
+        //display menu at the bottom
+        if (freeSpaceBottom > freeSpaceTop) {
```

Comment:
There are problems in the position and height calculation.
Offset should be taken into account every time, and the idea is that the content is never outside of the window but that is not the case.

You should consider separating the 2 concepts here : position and content height.
First determine the best position (considering offset and content height)
Then, if the content is still outside of the window (set a correct height to be at minimum at the offset of a border)